[plsql] ParseException when parsing VIEW

Please, prefix the report title with the language it applies to within brackets, such as [java] or [apex].
If not specific to a language, you can use [core].

**Affects PMD Version:** 
6.1.0
**Rule:**
Issue parsing a .sql file that has a CREATE OR REPLACE VIEW 
**Description:**
net.sourceforge.pmd.PMDException: Error while parsing D:\flOraGithubRepo\HR_WF\Database\HR\Views\EMP_DETAILS_VIEW.sql
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:98)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:50)
	at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:75)
	at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:23)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
Caused by: net.sourceforge.pmd.lang.plsql.ast.ParseException: Encountered "" at line 8, column 2.
Was expecting one of:
    "/" ...
    ";" ...
    
	at net.sourceforge.pmd.lang.plsql.ast.PLSQLParser.generateParseException(PLSQLParser.java:40789)
	at net.sourceforge.pmd.lang.plsql.ast.PLSQLParser.jj_consume_token(PLSQLParser.java:40643)
	at net.sourceforge.pmd.lang.plsql.ast.PLSQLParser.View(PLSQLParser.java:22737)
	at net.sourceforge.pmd.lang.plsql.ast.PLSQLParser.Input(PLSQLParser.java:182)
	at net.sourceforge.pmd.lang.plsql.PLSQLParser.parse(PLSQLParser.java:48)
	at net.sourceforge.pmd.SourceCodeProcessor.parse(SourceCodeProcessor.java:110)
	at net.sourceforge.pmd.SourceCodeProcessor.processSource(SourceCodeProcessor.java:173)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:95)
**Code Sample demonstrating the issue:**
Note that the SQL below was generated using Oracle SQL Developer so I'd have hoped that this is valid.

  CREATE OR REPLACE FORCE VIEW "HR_DEV1"."EMP_DETAILS_VIEW" ("EMPLOYEE_ID") AS 
  SELECT
  e.employee_id
FROM
  employees e
WITH READ ONLY
;



**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
windows command line: >call Tools\pmd\bin\pmd -d "Database\HR\Procedures,Database\HR\Functions,Database\HR\Packages,Database\HR\PackageBodies,Database\HR\Triggers,Database\HR\Views" -l plsql -R pmd_ruleset.xml -f summaryhtml -r Artifacts\static_analysis_report.html

--- a/docs/pages/release_notes.md
+++ b/docs/pages/release_notes.md
@@ -28,6 +28,8 @@ This is a minor release.
     *   [#1216](https://github.com/pmd/pmd/issues/1216): \[java] UnnecessaryFullyQualifiedName false positive for the same name method
 *   java-design
     *   [#1217](https://github.com/pmd/pmd/issues/1217): \[java] CyclomaticComplexityRule counts ?-operator twice
+*   plsql
+    *   [#981](https://github.com/pmd/pmd/issues/981): \[plsql] ParseException when parsing VIEW
 
 ### API Changes
 

--- a/pmd-plsql/etc/grammar/PldocAST.jjt
+++ b/pmd-plsql/etc/grammar/PldocAST.jjt
@@ -2503,8 +2503,7 @@ ASTView View()  :
   ["(" ViewColumn() ("," ViewColumn())* ")"]
   //### OF ... WITH OBJECT IDENTIFIER...
   <AS>
-  Statement() //SRT select()
-  //### WITH ...
+  SqlStatement(null,";")
   (";" | "/")
   
   { jjtThis.setImage(simpleNode.getImage()) ;  return jjtThis ; }

--- a/pmd-plsql/src/test/java/net/sourceforge/pmd/lang/plsql/ast/ViewTest.java
+++ b/pmd-plsql/src/test/java/net/sourceforge/pmd/lang/plsql/ast/ViewTest.java
@@ -1,0 +1,21 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.plsql.ast;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import net.sourceforge.pmd.lang.plsql.AbstractPLSQLParserTst;
+
+public class ViewTest extends AbstractPLSQLParserTst {
+
+    @Test
+    public void parseCreateViewIssue981() throws Exception {
+        String code = IOUtils.toString(this.getClass().getResourceAsStream("ViewIssue981.pls"));
+        ASTInput input = parsePLSQL(code);
+        Assert.assertNotNull(input);
+    }
+}

--- a/pmd-plsql/src/test/resources/net/sourceforge/pmd/lang/plsql/ast/ViewIssue981.pls
+++ b/pmd-plsql/src/test/resources/net/sourceforge/pmd/lang/plsql/ast/ViewIssue981.pls
@@ -1,0 +1,11 @@
+--
+-- BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+--
+
+CREATE OR REPLACE FORCE VIEW "HR_DEV1"."EMP_DETAILS_VIEW" ("EMPLOYEE_ID") AS
+SELECT
+e.employee_id
+FROM
+employees e
+WITH READ ONLY
+;

